File: python/disable_users.py
from looker_sdk import client, models


admin_client = client.setup("../../looker.ini")

# List of user ids to disable
user_ids = [100, 105, 106, 1800]

for user_id in user_ids:
    found = admin_client.search_users(id=user_id)
    if found:
        admin_client.update_user(user_id, models.WriteUser(is_disabled=True))
        print(f"user {user_id}: disabled")
    else:
        print(f"user {user_id}: not found")

File: examples/ruby/disable_users.rb
require 'looker-sdk'

# get API creds from environment variables
sdk = LookerSDK::Client.new(
  :client_id => ENV['LOOKERSDK_CLIENT_ID'],
  :client_secret => ENV['LOOKERSDK_CLIENT_SECRET'],
  :api_endpoint => ENV['LOOKERSDK_BASE_URL']
)

# list the users to disable in the instance
users_to_disable = [264, 214, 260]
body = {"is_disabled": true}


users_to_disable.each { |item|
    sdk.update_user(item, body)
    puts "User '#{sdk.user(item)[:first_name]} #{sdk.user(item)[:last_name]}' was disabled (id: #{item})"
}
